### ISSUE TITLE
Flavor-specific entities?

### ISSUE THREAD

**Issue by vostreltsov** [greenrobot/greenDAO#899]
So I have a few flavors and would like to have a bit different sets of entity properties in each of them (but keeping entity names the same)

If I put `@Entity` classes in each flavor, dao generator seems to ignore flavors and scan all the subdirs. As a result, in the generated DaoSession I see:
```
private final EntityConfig entityDaoConfig;
private final EntityConfig entityDaoConfig;
private final EntityConfig entityDaoConfig;
private final EntityConfig entityDaoConfig;
private final EntityConfig entityDaoConfig;
```

There are a lot more compilation errors.

Is there any way to make `@Entity` classes flavor-specific? Thanks.

**Comment by vostreltsov** [comment]
Ok, I see a similar issue #587. I have implemented a really dirty hack to get this working.

I edited my `build.gradle` with the following changes:
* make `tasks.greendao` dependent on a function which moves all `@Entity` classes from other flavors to a temp directory, so greendao generator works as expected. https://stackoverflow.com/a/34679933 describes how to get current flavor.
* make another task which depends on `tasks.greendao`, and which `tasks."generate${flavor.name.capitalize()}DebugSources"` depend on. This task is to move back all the files to their initial location.

This really worked for me, hope this helps somebody else :)
